### ISSUE TITLE
async not dropped from delegate keyword

### ISSUE THREAD

**Issue by persn** [zompinc/sync-method-generator#64]
Consider the following code

```csharp
[Zomp.SyncMethodGenerator.CreateSyncVersion]
public async Task FooAsync()
{
    Bar(async () => { await Task.Delay(100); });

    Bar(async delegate () { await Task.Delay(100); });

    await Task.Delay(100);
}

public void Bar(Action action) { }
```

Will generate the following code
```csharp
public void Foo()
{
    Bar(() => { global::System.Threading.Thread.Sleep(100); });

    Bar(async delegate () { global::System.Threading.Thread.Sleep(100); });

    global::System.Threading.Thread.Sleep(100);
}
```

As we can see the async keyword is not dropped from before the delegate keyword. This will give `warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.`, which is a warning for me since my company has enabled werror

**Comment by virzak** [comment]
@persn The recent issues you filed are available in version 1.3.33.

**Comment by persn** [comment]
@virzak I tested them out and they seem to work. Thanks
